### PR TITLE
Alarm / Threshold updates (and some other UI updates...)

### PR DESCRIPTION
**Pull request by jasoncalabrese**
Many users don't understand the current Loss/PredictAR based alarms, they also want to be able to set their own target range instead of the default 80-180.

This PR aims to make the target range and alarm types configurable, this should address most of #271.  To do this 5 new ENV variables have been added.

   * `BG_HIGH` - default: **260**, the high BG outside the target range that is considered urgent
   * `BG_TARGET_TOP` - default: **180**, the top of the target range, also used to draw the line on the chart
   * `BG_TARGET_BOTTOM` - default: **80**, the bottom of the target range, also used to draw the line on the chart
   * `BG_LOW` - default: **55**, the low BG outside the target range that is considered urgent
   * `ALARM_TYPES` - default to **simple** if any `BG_`* ENV's are set, otherwise default to **predict**.  There currently 2 alarm types are supported, and can be used independently are separately.  The `simple` alarm type only compares the current BG to `BG_` thresholds above, the `predict` alarm type works as before by using a highly tuned formula that forecasts where the BG is going based on it's trend.

### CODE REVIEW

File: static/css/main.css [hunk 1]
```diff
@@ -238,6 +242,11 @@ div.tooltip {
     font-size: 70%;
   }
 
+  html body #bgButton {
```
Comment (trhodeos) [inline static/css/main.css]: `html body` is unnecessary.

Comment (trhodeos) [inline static/css/main.css]: Also, should this be merged with the `#bgButton` rule on line 153?

File: static/js/client.js [hunk 2]
```diff
@@ -436,6 +405,17 @@
         // selects all our data into data and uses date function to get current max date
         var focusCircles = focus.selectAll('circle').data(focusData, dateFn);
 
+        var focusDotRadius = function(d) {
+            if (d.type == 'mbg')
+                return 6;
+            else if ($(window).width() > WIDTH_BIG_DOTS)
```
Comment (trhodeos) [inline static/js/client.js]: cache $(window) into a local var.

File: static/js/client.js [hunk 3]
```diff
@@ -452,8 +433,30 @@
...
+                tooltip.html('<strong>' + bgType + ' BG:</strong> ' + d.sgv +
+                    (d.type == 'mbg' ? '<br/><strong>Device: </strong>' + d.device : '') +
+                    '<br/><strong>Time:</strong> ' + formatTime(d.date))
+                    .style("left", (d3.event.pageX) + "px")
+                    .style("top", (d3.event.pageY - 28) + "px");
+            })
+            .on('mouseout', function (d) {
+                if (d.type != "sgv" && d.type != 'mbg') return;
+                tooltip.transition()
+                    .duration(500)
```
Comment (trhodeos) [inline static/js/client.js]: any reason why the mouseout and mouseover durations are different?

File: static/js/client.js [hunk 4]
```diff
@@ -462,11 +465,12 @@
         d3.selectAll('.path').remove();
 
         // add treatment bubbles
-        //
-        //var bubbleSize = prevChartWidth < 400 ? 4 : (prevChartWidth < 600 ? 3 : 2);
-        //focus.selectAll('circle')
-        //    .data(treatments)
-        //    .each(function (d) { drawTreatment(d, bubbleSize, true) });
+
+        var bubbleSize = prevChartWidth < 400 ? 4 : (prevChartWidth < 600 ? 3 : 2);
```
Comment (trhodeos) [inline static/js/client.js]: can this use the `focusDotRadius` function (or a helper function that both use)? This logic seems similar.

File: static/js/client.js [hunk 5]
```diff
@@ -505,39 +516,33 @@
...
-                  .attr('x', function (d) { return xScale(d.created_at); })
-                  .attr('y', function (d) { return yScale(scaleBg(500)); })
-                  .attr("width", 15)
-                  .attr("height", 15)
-                  .attr("rx", 6)
-                  .attr("ry", 6)
+            treatCircles.enter().append('circle')
+                  .attr('cx', function (d) { return xScale(d.created_at); })
+                  .attr('cy', function (d) { return yScale(scaleBg(d.glucose || calcBGByTime(d.created_at.getTime()))); })
+                  .attr("r", 6)
```
Comment (trhodeos) [inline static/js/client.js]: Do we want this to be 6 always, regardless of screen size?

File: static/js/client.js [hunk 6]
```diff
@@ -545,11 +550,11 @@
...
                   })
-          .on("mouseout", function (d) {
-              div.transition()
-                  .duration(500)
-                  .style("opacity", 0);
-          });
+                  .on('mouseout', function (d) {
+                      tooltip.transition()
+                          .duration(500)
+                          .style("opacity", 0);
```
Comment (trhodeos) [inline static/js/client.js]: Might as well change these to be single quotes as well.

File: static/js/client.js [hunk 7]
```diff
@@ -1133,33 +1004,49 @@
 
     }
 
-    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    function calcBGByTime(time) {
+        var closeBGs = data.filter(function(d) {
+            if (!d.y) return false;
```
Comment (trhodeos) [inline static/js/client.js]: surround with `{}`s

**Comment by bewest** [comment]
Looking good.

**Comment by trhodeos** [comment]
Added a few comments. Looks good otherwise.

**Comment by jasoncalabrese** [comment]
@trhodeos thanks for the code review, think I took care of those issues

I merged this to my prod branch and will be using it all day tomorrow, if there aren't any new issues I plan to merge tomorrow.

**Pull request merged**